cas-server-webapp-actions/src/main/java/org/jasig/cas/web/flow/AuthenticationViaFormAction.java

```diff
@@ -165,14 +164,13 @@ protected Event grantServiceTicket(final RequestContext context, final Credentia
 
         } catch (final AuthenticationException e) {
             return newEvent(AUTHENTICATION_FAILURE, e);
-        } catch (final TicketCreationException e) {
-            logger.warn("Invalid attempt to access service using renew=true with different credential. Ending SSO session.");
-            this.centralAuthenticationService.destroyTicketGrantingTicket(ticketGrantingTicketId);
         } catch (final AbstractTicketException e) {
-            return newEvent(AbstractCasWebflowConfigurer.TRANSITION_ID_ERROR, e);
+            if (e instanceof TicketCreationException) {
+                logger.warn("Invalid attempt to access service using renew=true with different credential. Ending SSO session.");
```

Comment:
This error is a little too specific. Are we sure it's appropriate for all TicketCreationException instances?